packages/core-mobile/app/new/routes/RootNavigator.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import {
+  forNoAnimation,
+  stackNavigatorScreenOptions
+} from 'common/consts/screenOptions'
+import { useModalScreenOptions } from 'common/hooks/useModalScreenOptions'
+import { useWalletState } from 'contexts/WalletStateContext'
+import { Stack } from 'common/components/Stack'
+import { WalletState } from 'store/app/types'
+
+export function RootNavigator(): JSX.Element {
+  const { isLocked, walletState } = useWalletState()
+  const { modalScreensOptions } = useModalScreenOptions()
+
+  return (
+    <Stack
+      screenOptions={{
+        ...stackNavigatorScreenOptions,
+        headerShown: false
+      }}>
+      {/* verified and wallet active */}
+      <Stack.Protected guard={!isLocked && walletState === WalletState.ACTIVE}>
+        <Stack.Screen
+          name="(signedIn)"
```

Comment:
is it necessary to check `isLocked` in this navigator? when isLocked is true, wallet state should be inactive and when it is false, wallet state should be active.